dotnet/secrethub.i

```diff
@@ -0,0 +1,82 @@
+%module Client
```

Comment:
An open question is whether we want to call this `Client` or something like `SimpleClient` or `LiteClient` to indicate that it does not have to full functionality of the SecretHub client.

@florisvdg @SimonBarendse any thoughts on this?

Reply:
What does the migration path look like **if** we want to add more functionality? Can we easily keep the existing functionality available at `LiteClient`, while we add more functionality in `Client`?

I'm not sure if it's necessary to indicate the incompleteness in the import path itself. I've seen many SDKs indicating supported endpoints/operations in the readme of the repo. We could adopt a similar approach.

Reply:
One of the consideration that I did not make clear in my message is the way this client is initialized: it can only be initialized with default options and is not a full-fledged client in that way. But that's also something that we can try to fix (see https://github.com/secrethub/secrethub-xgo/pull/16#discussion_r488655898)

Reply:
But is it then even a client if it's all managed and created for you under the hood anyway? Shouldn't it just be `SecretHub.Read` in that case?

Because if we bring the fact that it's a `Client` to the forefront, you may expect more flexibility which is not there at the moment.

Reply:
And when we do add a more powerful client, it will still be nice to have a basic `SecretHub.Read`, so we won't even have to deprecate that.

Reply:
Maybe that's the better approach here. Though I am not sure whether that is technically feasible. @Marton6 probably knows that best.

Reply:
I like that idea @florisvdg

Reply:
I changed the code generation such that now all functions are methods on the `Client` object:
```c#
var client = new SecretHub.Client();
string secret = client.ReadString("secrethub-xgo/dotnet/test-secret");
```

This is forward compatible with adding client options and more methods, so I'd say it's safe to call it `Client` (the module name is different now but it does not matter). So I suggest we have a `SecretHub` namespace with a `Client` class in it.

Reply:
That sounds good to me